ballista/core/src/serde/mod.rs

```diff
@@ -291,8 +294,11 @@ impl PhysicalExtensionCodec for BallistaPhysicalExtensionCodec {
                 )?))
             }
             PhysicalPlanType::ShuffleReader(shuffle_reader) => {
+                let default_codec =
```

Comment:
would it make sense to make default_codec property `BallistaPhysicalExtensionCodec`, so it can be overridden if needed? `BallistaLogicalExtensionCodec` does similar thing. (also there is few other places where `DefaultPhysicalCodec` is created)

Reply:
good point, thanks!